package/MDAnalysis/coordinates/TRR.py

```diff
@@ -151,6 +152,28 @@ class TRRReader(XDRBaseReader):
     _writer = TRRWriter
     _file = TRRFile
 
+    def _read_next_timestep(self, ts=None):
+        """copy next frame into timestep
+        
+        versionadded:: 2.4.0
+            TRRReader implements this method so that it can use
+            read_direct_xvf to read the data directly into the timestep
+            rather than copying it from a temporary array.
+        """
+        if self._frame == self.n_frames - 1:
+            raise IOError(errno.EIO, 'trying to go over trajectory limit')
+        if ts is None:
+            ts = self.ts
```

Comment:
Can we raise an issue about this `ts=None` thing and remove it for 3.0.. it's not clear when this is ever used.  I imagine this is related to the copy behaviour confusion too. I think a Reader should hold a `Timestep` which is reused consistently, then people can **manually** copy if they want to.

Reply:
Issue raised #3928